### ISSUE TITLE
Add Catalog filename pattern not working as expected for multi discs

### ISSUE THREAD

**Issue by xekon** [ampache/ampache#2295]
## Description
Add Catalog filename pattern not working

### Describe the bug

I used these auto-inserted fields when adding my catalog: 

filename pattern: %d_%T - %t
folder pattern: %a/%y - %A

My music is structured like this:

/audio/artist/Year - Album/01 - The First Song.mp3
/audio/artist/Year - Album/02 - The Second Song.mp3

however for multi disc albums it is like this:

/audio/artist/Year - Album/1_01 - The First Song on CD1.mp3
/audio/artist/Year - Album/1_02 - The Second Song on CD1.mp3
/audio/artist/Year - Album/2_01 - The First Song on CD2.mp3
/audio/artist/Year - Album/2_02 - The Second Song on CD2.mp3

Unfortunately with the filename pattern that I used ONLY the multi disc albums are properly indexed. For albums without a disc number they all end up in a single Orphaned album.

I was hoping maybe there is a way that makes %d_ optional so that it can use it if that part of the pattern is present, otherwise ignore it. I am not seeing any such option though.

Also I am 100% certain that all songs and directory structure are conforming, I used find with grep and regex to make sure:

`find -type f -name "*.mp3" | grep -vP "^\./[a-zA-Z0-9 ]+/[0-9]{4} - [a-zA-Z0-9 \[\]\(\)',&–=#]+/([0-9]{2}|[0-9]+_[0-9]{2}) - [a-zA-Z0-9 \[\]\(\)\!'.,&–=#+]+\.mp3" | head -10`

**If the field supported some basic regex I could do something like this:** `(%d_|)%T - %t`

### Expected behavior
be able to make use of the %d for disk number for albums that are multi disc, and still have single disc albums index correctly.

### Screenshots
![image](https://user-images.githubusercontent.com/106644/74221973-29cf9080-4c68-11ea-9362-2a6795ee2b66.png)

![image](https://user-images.githubusercontent.com/106644/74222112-86cb4680-4c68-11ea-8386-3fdaffd74950.png)

## Environment

* Ampache version: 4.2.0-develop
* Web server + version: nginx latest
* Server operating system: Ubuntu 18.04.4
* Client operating system: Ubuntu 18.04.4 (Firefox latest)

**Comment by lachlan-00** [comment]
so these files don't have tags and Ampache falls back to the file/folder pattern?

I didn't know this actually happened outside bin/sort_files.inc

**Comment by xekon** [comment]
correct, these files do not have any tags at all.... I suppose I could use a filename to tag tool that mass tags the files based on filename and path if that is all that it takes to remedy this hangup.

**Comment by AshotN** [comment]
Have you tried [Picard](https://picard.musicbrainz.org/) to tag your files for you? Works really well.

**Comment by xekon** [comment]
I did use Picard and it sometimes would label an album AC DC and other times ac dc or ac/dc the resulting variations would cause Ampache to display them as different artists or albums. Ultimately I found it very Frustrating, because I am coming from using Airsonic, and they must do some sort of magic to see that the files are in the same folder and thus the same album, because I did not experience that weirdness there.

However ampache has 5 star music ratings PER Song which is exactly what I want, so I am stubbornly trying to get Ampache working the way I want so that I can use it instead of Airsonic. So I figured fine I will just nuke all of the Tags! which I did lol! I guess now I need to add the tags back based on filename and path, but I dont think I will be using Picard.

**Comment by AshotN** [comment]
Picard's database is editable, it uses ID's that are unique to artists to avoid the issue you are mentioning. It may be tedious to have to fix all that, but it will save someone in the future the pain and will help you out as well. 

Correct me if I'm wrong, as I am not familiar with Airsonic's logic, but I could see an issue with processing songs in the same folder as part of the same album. If you had a folder named something like Armenian music for example. 

Ultimately, proper tagging may be annoying, but I think it's the only sound way to expect any kind of proper cataloging. Making a bunch of patches that assume the correct answer will just lead to a lot of weird edge cases.

**Comment by xekon** [comment]
Yeah I appreciate the responses, I am using Puddletag to add the tags from filename now. I already painstakingly made sure that all filename and paths are correct so I may as well use them.

**Comment by lachlan-00** [comment]
i think it would be a good fallback especially for things like wav files. Right now i convert wav to flac but it would be good to keep the original files.

**Comment by lachlan-00** [comment]
i think getting onto it now @xekon is a good idea. we have a ton of features to work on but i think we should fall back when something matches our path (it will just take a while)

**Comment by AshotN** [comment]
As long as it's a last ditch resort I think it shouldn't cause any problems. Good luck @xekon!

**Comment by xekon** [comment]
This is no longer an issue for me simply because I now make sure that all multi disc albums include the disk number id3 tag.
